### PR TITLE
Improve Python CLI OTM validator

### PR DESCRIPTION
**Pull request by dfernandezvigo**

### CODE REVIEW

File: startleft/iriusrisk.py [hunk 1]
```diff
@@ -230,3 +230,76 @@ def validate_otm(self):
...
+        wrong_trustzone_ids = []
+        wrong_component_ids = []
+        wrong_component_parent_ids = []
+        wrong_dataflow_ids = []
+        wrong_dataflow_from_ids = []
+        wrong_dataflow_to_ids = []
+
+        for trustzone in self.otm['trustzones']:
+            if trustzone['id'].strip() == '':
+                wrong_trustzone_ids.append(trustzone['id'])
```
Comment (DavidRodriguez-IriusRisk) [inline startleft/iriusrisk.py:246]: Here we are adding empty values to the wrong_trustzone_ids, which won't be a good error message to log. Also I think that validations of required parameters as ids is currently done in the schema validation, so probably we shouldn't have to do it here at all

(Same comment components and data flows)

File: startleft/iriusrisk.py [hunk 2]
```diff
@@ -230,3 +230,76 @@ def validate_otm(self):
...
+                all_valid_ids.append(dataflow['id'])
+
+            if dataflow['from'] not in all_valid_ids:
+                wrong_dataflow_from_ids.append(dataflow['from'])
+
+            if dataflow['to'] not in all_valid_ids:
+                wrong_dataflow_to_ids.append(dataflow['to'])
+
+
+        if len(wrong_trustzone_ids) > 0:
```
Comment (DavidRodriguez-IriusRisk) [inline startleft/iriusrisk.py:278]: All "len(array) > 0" can be replaced by "not array" according to Python style guides: https://www.python.org/dev/peps/pep-0008/
It's just a suggestion, change it only if you also prefer that notation

File: startleft/iriusrisk.py [hunk 3]
```diff
@@ -230,3 +230,76 @@ def validate_otm(self):
...
+        if len(wrong_dataflow_from_ids) > 0:
+            logger.error(f"Dataflow 'from' identifiers inconsistent: {wrong_dataflow_from_ids}")
+
+        if len(wrong_dataflow_to_ids) > 0:
+            logger.error(f"Dataflow 'to' identifiers inconsistent: {wrong_dataflow_to_ids}")
+
+        if len(repeated_ids) > 0:
+            logger.error(f"Repeated identifiers inconsistent: {repeated_ids}")
+
+        if len(wrong_trustzone_ids) > 0 or len(wrong_component_ids) > 0 or len(wrong_component_parent_ids) > 0\
```
Comment (DavidRodriguez-IriusRisk) [inline startleft/iriusrisk.py:299]: Maybe returning directly the expression instead a true or false booleans is more readable.

`return wrong_component_parent_ids or wrong_dataflow_from_ids or wrong_dataflow_to_ids or repeated_ids`

File: startleft/iriusrisk.py [hunk 4]
```diff
@@ -230,3 +230,76 @@ def validate_otm(self):
...
+            elif dataflow['id'] not in all_valid_ids:
+                all_valid_ids.append(dataflow['id'])
+
+            if dataflow['from'] not in all_valid_ids:
+                wrong_dataflow_from_ids.append(dataflow['from'])
+
+            if dataflow['to'] not in all_valid_ids:
+                wrong_dataflow_to_ids.append(dataflow['to'])
+
+
```
Comment (DavidRodriguez-IriusRisk) [inline startleft/iriusrisk.py:277]: Please remove one of these empty line breaks

**Review by DavidRodriguez-IriusRisk — CHANGES_REQUESTED** [review]

File: startleft/iriusrisk.py (hunk 1 above)
Comment (DavidRodriguez-IriusRisk) [reply startleft/iriusrisk.py:246]: I've just checked the schema and it doesn't fails on empty fields, it only fails when the node doesn't exists. Probably is better to adapt the schema validation to also fail on these situations.

Comment (dfernandezvigo) [reply startleft/iriusrisk.py:246]: Changes done.

File: startleft/iriusrisk.py (hunk 3 above)
Comment (dfernandezvigo) [reply startleft/iriusrisk.py:299]: done

File: startleft/iriusrisk.py (hunk 2 above)
Comment (dfernandezvigo) [reply startleft/iriusrisk.py]: exactly. It has been replaced "len(sth)>0:" by "len(sth):"

**Review by DavidRodriguez-IriusRisk — APPROVED** [review]

**Pull request merged**